vm/systemSmartContracts/staking.go

```diff
@@ -614,6 +635,50 @@ func (r *stakingSC) unStake(args *vmcommon.ContractCallInput) vmcommon.ReturnCod
 	return vmcommon.Ok
 }
 
+func (r *stakingSC) moveFirstFromWaitingToStakedIfNeeded(blsKey []byte) error {
+	waitingElementKey := r.createWaitingListKey(blsKey)
+	elementInList, err := r.getWaitingListElement(waitingElementKey)
+	if err == nil {
+		// node in waiting - remove from it - and that's it
+		return r.removeFromWaitingList(blsKey)
+	}
+
+	waitingList, err := r.getWaitingListHead()
+	if err != nil {
+		return err
+	}
+	if waitingList.Length == 0 {
+		return nil
+	}
+	elementInList, err = r.getWaitingListElement(waitingList.FirstKey)
+	if err != nil {
+		return err
+	}
+	err = r.removeFromWaitingList(elementInList.BLSPublicKey)
+	if err != nil {
+		return err
+	}
+
+	nodeData, err := r.getOrCreateRegisteredData(elementInList.BLSPublicKey)
+	if err != nil {
+		return err
+	}
+	if len(nodeData.RewardAddress) == 0 || nodeData.Staked {
+		return vm.ErrInvalidWaitingList
+	}
+
+	nodeData.Waiting = false
+	nodeData.Staked = true
+	nodeData.StakedNonce = r.eei.BlockChainHook().CurrentNonce()
+	stakeValue := r.getStakeValueForCurrentEpoch()
+	if nodeData.StakeValue.Cmp(stakeValue) < 0 {
```

Comment:
is this compare correct? If stakeValue < currentEpochStakeValue set the node.StakeValue to the stakeValue from the current Epoch?

Reply:
this will change when auction is working - the idea is that selected nodes will have selected stake value.

Reply:
ok